src/Compilers/Test/Utilities/CSharp/TestOptions.cs

```diff
@@ -47,34 +47,34 @@ public static class TestOptions
         public static readonly CSharpParseOptions RegularWithoutPatternCombinators = Regular8;
         public static readonly CSharpParseOptions RegularWithPatternCombinators = RegularPreview;
 
-        public static readonly CSharpCompilationOptions ReleaseDll = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, optimizationLevel: OptimizationLevel.Release);
-        public static readonly CSharpCompilationOptions ReleaseExe = new CSharpCompilationOptions(OutputKind.ConsoleApplication, optimizationLevel: OptimizationLevel.Release);
+        public static readonly CSharpCompilationOptions ReleaseDll = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, optimizationLevel: OptimizationLevel.Release).WithWarningLevel(5);
```

Comment:
Consider extracting a private helper like the following

```cs
private void CreateTestOptions(OutputKind outputKind, OptimizationLevel optimizationLevel) => new CSharpCompilationOptions(outputKind, optimizationLevel: optimizationLevel, warningLevel: 9999)
```
Since we have documented that 9999 means "always use the latest", this should make it so our tests also start issuing the latest warnings whenever they happen to get added.

Reply:
Rather than use 9999 directly (aka magic number) would prefer we define a named constant along the lines of `MaxWarningLevel` that has this value and use that instead.